src/main/java/cn/nukkit/block/BlockGlowLichen.java

```diff
@@ -0,0 +1,269 @@
+
+        if (!target.isSolid() && target.getId() != BlockID.GLOW_LICHEN) {
+            return false;
+        }
+
+        int currentMeta = 0;
+        if (block.getId() == GLOW_LICHEN) {
+            currentMeta = block.getPropertyValue(MULTI_FACE_DIRECTION_BITS);
+        }
+
+        setPropertyValue(MULTI_FACE_DIRECTION_BITS, currentMeta | (0b0001 << face.getOpposite().getIndex()));
+
+        if (getPropertyValue(MULTI_FACE_DIRECTION_BITS) == currentMeta) {
+            BlockFace[] sides = BlockFace.values();
+            Stream<BlockFace> faceStream = Arrays.stream(sides).filter(side ->
+                    block.getSide(side).isSolid() && !isGrowthToSide(side)
+            );
+            Optional<BlockFace> optionalFace = faceStream.findFirst();
+            if (optionalFace.isPresent()) {
+                growToSide(optionalFace.get());
+                return true;
+            }
+
+            return false;
+        }
+
+        getLevel().setBlock(block, this, true, true);
+        return true;
+    }
+
+    @Override
+    public boolean onActivate(@Nonnull Item item, @Nullable Player player) {
+
+        if (!item.isFertilizer()) {
+            return false;
+        }
+
+        Map<Block, BlockFace> candidates = new HashMap<>();
+
+        for (int i = 0; i < 5; i++) {
+            BlockFace side = BlockFace.fromIndex(i);
+            Block support = this.getSide(side);
+
+            if (isGrowthToSide(side)) {
+                BlockFace[] supportSides = side.getEdges().toArray(new BlockFace[0]);
+
+                for (BlockFace supportSide : supportSides) {
+                    Block supportNeighbor = support.getSide(supportSide);
+
+                    if (supportNeighbor.getId() == BlockID.AIR) {
+                        candidates.put(supportNeighbor, supportSide.getOpposite());
+                    }
+
+                    if (!supportNeighbor.isSolid()) {
+                        continue;
+                    }
+
+                    Block supportNeighborOppositeSide = supportNeighbor.getSide(side.getOpposite());
+                    if (supportNeighborOppositeSide.getId() == BlockID.AIR || supportNeighborOppositeSide.getId() == BlockID.GLOW_LICHEN) {
+                        if (supportNeighborOppositeSide.getId() == BlockID.GLOW_LICHEN &&
+                                (((BlockGlowLichen) supportNeighborOppositeSide).isGrowthToSide(side)||
+                                supportNeighborOppositeSide.getSide(side).getId() == BlockID.AIR)) {
+                            continue;
+                        }
+                        candidates.put(supportNeighborOppositeSide, side);
+                    }
+
+                }
+
+            } else {
+                if (support.isSolid()) {
+                    candidates.put(this, side);
+                }
+            }
+        }
+
+        if (!candidates.isEmpty()) {
```

Comment:
I recommend inverting this condition, seen that an empty list is the base case. That'd remove an unneeded layer of indentation.